{
	"acknowledgement": "Red Hat would like to thank Damien Aumaitre (Quarkslab) and Nicolas Surbayrole (Quarkslab) for reporting this issue.",
	"affected_release": [
		{
			"advisory": "RHBA-2020:0547",
			"cpe": "cpe:/a:redhat:ansible_tower:3.4::el7",
			"package": "ansible-tower-34/ansible-tower-memcached:1.4.15-28",
			"product_name": "Red Hat Ansible Tower 3.4 for RHEL 7",
			"release_date": "2020-02-18T00:00:00Z"
		},
		{
			"advisory": "RHBA-2020:0547",
			"cpe": "cpe:/a:redhat:ansible_tower:3.4::el7",
			"package": "ansible-tower-35/ansible-tower-memcached:1.4.15-28",
			"product_name": "Red Hat Ansible Tower 3.4 for RHEL 7",
			"release_date": "2020-02-18T00:00:00Z"
		},
		{
			"advisory": "RHBA-2020:0547",
			"cpe": "cpe:/a:redhat:ansible_tower:3.4::el7",
			"package": "ansible-tower-37/ansible-tower-memcached-rhel7:1.4.15-28",
			"product_name": "Red Hat Ansible Tower 3.4 for RHEL 7",
			"release_date": "2020-02-18T00:00:00Z"
		},
		{
			"advisory": "RHBA-2020:1539",
			"cpe": "cpe:/a:redhat:ansible_tower:3.5::el7",
			"package": "ansible-tower-35/ansible-tower:3.5.6-1",
			"product_name": "Red Hat Ansible Tower 3.5 for RHEL 7",
			"release_date": "2020-04-22T00:00:00Z"
		}
	],
	"bugzilla": {
		"description": "CVE-2020-1738 ansible: module package can be selected by the ansible facts",
		"id": "1802164",
		"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1802164"
	},
	"csaw": false,
	"cvss3": {
		"cvss3_base_score": "3.9",
		"cvss3_scoring_vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:C/C:N/I:L/A:L",
		"status": "verified"
	},
	"cwe": "CWE-88",
	"details": [
		"A flaw was found in Ansible Engine when the module package or service is used and the parameter 'use' is not specified. If a previous task is executed with a malicious user, the module sent can be selected by the attacker using the ansible facts file. All versions in 2.7.x, 2.8.x and 2.9.x branches are believed to be vulnerable.",
		"A flaw was found in Ansible Engine when the module package or service is used and the parameter 'use' is not specified. If a previous task is executed with a malicious user, the module sent can be selected by the attacker using the ansible facts file."
	],
	"mitigation": {
		"lang": "en:us",
		"value": "Specify the parameter 'use' when possible on the package and service modules. Avoid using Ansible Collections on Ansible 2.8.9 or 2.7.16 (and any of the previous versions) as they are not rejecting python with no path (already fixed in 2.9.x)."
	},
	"name": "CVE-2020-1738",
	"package_state": [
		{
			"cpe": "cpe:/a:redhat:cloudforms_managementengine:5",
			"fix_state": "Not affected",
			"package_name": "ansible-tower",
			"product_name": "CloudForms Management Engine 5"
		},
		{
			"cpe": "cpe:/a:redhat:ansible_engine:2",
			"fix_state": "Will not fix",
			"package_name": "ansible",
			"product_name": "Red Hat Ansible Engine 2"
		},
		{
			"cpe": "cpe:/a:redhat:ansible_tower:3",
			"fix_state": "Will not fix",
			"package_name": "ansible",
			"product_name": "Red Hat Ansible Tower 3"
		},
		{
			"cpe": "cpe:/a:redhat:ceph_storage:2",
			"fix_state": "Out of support scope",
			"package_name": "ansible",
			"product_name": "Red Hat Ceph Storage 2"
		},
		{
			"cpe": "cpe:/a:redhat:ceph_storage:3",
			"fix_state": "Will not fix",
			"package_name": "ansible",
			"product_name": "Red Hat Ceph Storage 3"
		},
		{
			"cpe": "cpe:/a:redhat:openstack:10",
			"fix_state": "Out of support scope",
			"package_name": "ansible",
			"product_name": "Red Hat OpenStack Platform 10 (Newton)"
		},
		{
			"cpe": "cpe:/a:redhat:openstack:13",
			"fix_state": "Will not fix",
			"package_name": "ansible",
			"product_name": "Red Hat OpenStack Platform 13 (Queens)"
		},
		{
			"cpe": "cpe:/a:redhat:storage:3",
			"fix_state": "Will not fix",
			"package_name": "ansible",
			"product_name": "Red Hat Storage 3"
		}
	],
	"public_date": "2020-02-18T00:00:00Z",
	"references": [
		"https://www.cve.org/CVERecord?id=CVE-2020-1738\nhttps://nvd.nist.gov/vuln/detail/CVE-2020-1738"
	],
	"statement": "Ansible Engine 2.7.16, 2.8.10, and 2.9.6 as well as previous versions are affected.\nAnsible Tower 3.4.5, 3.5.5 and 3.6.3 as well as previous versions are affected.\nIn Red Hat OpenStack Platform, because the flaw has a lower impact,  ansible is not directly customer exposed, and the fix would require a substantial amount of development, no update will be provided at this time for the RHOSP ansible package.",
	"threat_severity": "Low",
	"upstream_fix": "ansible-engine 2.9.7"
}
